tb_plugin/torch_tb_profiler/profiler/step_parser.py

```diff
@@ -0,0 +1,195 @@
+import sys
+from enum import IntEnum
+
+from .. import utils
+from .trace import EventTypes
+
+logger = utils.get_logger()
+
+ProfileRole = IntEnum('ProfileRole', ['Kernel', 'Memcpy', 'Memset', 'Communication', 'Runtime', 'DataLoader', 'CpuOp', 'Other', 'Total'], start=0)
+
+class StepParser:
+    def __init__(self):
+        # we could not use [[]] * len here since they all point to same memory
+        # https://stackoverflow.com/questions/12791501/python-initializing-a-list-of-lists
+        # https://stackoverflow.com/questions/240178/list-of-lists-changes-reflected-across-sublists-unexpectedly
+        self.role_ranges = [[] for _ in range(ProfileRole.Total - 1)]
+        self.steps = []
+        self.steps_names = []
+        self.min_ts = sys.maxsize
+        self.max_ts = -sys.maxsize - 1
+
+    def parse_events(self, events, node_parser):
+        for event in events:
+            self._parse_event(event, node_parser.communication_data)
+
+        if len(self.steps) == 0:
+            self.steps.append((self.min_ts, self.max_ts))
+            self.steps_names.append("0")
+        self._update_steps_consider_device_side(node_parser.runtime_node_list, node_parser.device_node_list)
```

Comment:
Why not put runtime_node_list and device_node_list into NodeContext. I think NodeContext is used to maintain the common data that will be used in multiple parsers, and each parser only maintain the data only used by themself?